src/rpc/blockchain.cpp

```diff
@@ -1167,6 +1168,141 @@ static RPCHelpMan pruneblockchain()
+                      "muhash,"
+                      "total_amount,"
+                      "total_unspendable_amount,"
+                      "total_prevout_spent_amount,"
+                      "total_new_outputs_ex_coinbase_amount,"
+                      "total_coinbase_amount,"
+                      "total_subsidy,"
+                      "total_genesis_block,"
+                      "total_bip30,"
+                      "total_unspendable_scripts,"
+                      "total_unclaimed_rewards,"
+                      "txs,"
+                      "timestamp,"
+                      "mediantime,"
+                      "chainwork,"
+                      "version\n");
+
+    CCoinsStats stats{CoinStatsHashType::MUHASH};
+    NodeContext& node = EnsureAnyNodeContext(request.context);
+    ChainstateManager& chainman = EnsureChainman(node);
+    CChainState& active_chainstate = chainman.ActiveChainstate();
+    active_chainstate.ForceFlushStateToDisk();
+    CBlockIndex *pindex = active_chainstate.m_chain.Genesis();
+
+    CCoinsView* coins_view;
+    BlockManager* blockman;
+    {
+        LOCK(::cs_main);
+        coins_view = &active_chainstate.CoinsDB();
+        blockman = &active_chainstate.m_blockman;
+    }
+
+    CCoinsStats prev_stats{CoinStatsHashType::MUHASH};
+
+    for (; pindex; pindex = active_chainstate.m_chain.Next(pindex)) {
+        if (!GetUTXOStats(coins_view, *blockman, stats, node.rpc_interruption_point, pindex))
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Unable to read UTXO set");
+
+        file << strprintf("%d,", static_cast<int64_t>(stats.nHeight));
+        file << strprintf("%s,", stats.hashBlock.GetHex());
+        file << strprintf("%d,", static_cast<int64_t>(stats.nTransactionOutputs));
+        file << strprintf("%d,", static_cast<int64_t>(stats.nBogoSize));
+        file << strprintf("%d,", stats.hashSerialized.GetHex());
+
+        CHECK_NONFATAL(stats.total_amount.has_value());
+        file << strprintf("%s,", FormatMoney(stats.total_amount.value()));
+
+        if (cummulative) {
+            file << strprintf("%s,", FormatMoney(stats.total_unspendable_amount));
+            file << strprintf("%s,", FormatMoney(stats.total_prevout_spent_amount));
+            file << strprintf("%s,", FormatMoney(stats.total_new_outputs_ex_coinbase_amount));
+            file << strprintf("%s,", FormatMoney(stats.total_coinbase_amount));
+            file << strprintf("%s,", FormatMoney(stats.total_subsidy));
+            file << strprintf("%s,", FormatMoney(stats.total_unspendables_genesis_block));
+            file << strprintf("%s,", FormatMoney(stats.total_unspendables_bip30));
+            file << strprintf("%s,", FormatMoney(stats.total_unspendables_scripts));
+            file << strprintf("%s,", FormatMoney(stats.total_unspendables_unclaimed_rewards));
+        } else {
+            file << strprintf("%s,", FormatMoney(stats.total_unspendable_amount - prev_stats.total_unspendable_amount));
```

Comment:
Nit: maybe calculate these values before the `if (cummulative)` to remove the repetition here:
```cpp
const unspendable_amount = tats.total_unspendable_amount - cumulative ? 0 : prev_stats.total_prevout_spent_amount
...
file << strprintf("%s,", FormatMoney(unspendable_amount));
```

Reply:
done